sources/packages/backend/apps/api/src/route-controllers/application/application.aest.controller.ts

```diff
@@ -183,73 +203,111 @@ export class ApplicationAESTController extends BaseController {
+      );
+    }
     return this.applicationControllerService.getInProgressApplicationDetails(
-      applicationId,
+      currentApplicationId,
     );
   }
 
   /**
    * Get status of all requests and confirmations in student application (Exception, PIR and COE).
-   * @param applicationId application id.
+   * @param applicationId current application id.
    * @returns application progress details.
    */
   @ApiNotFoundResponse({
-    description: "Application not found.",
+    description:
+      "Application not found or current application for provided parent application not found.",
   })
   @Get(":applicationId/progress-details")
   async getApplicationProgressDetails(
     @Param("applicationId", ParseIntPipe) applicationId: number,
   ): Promise<ApplicationProgressDetailsAPIOutDTO> {
+    const currentApplicationId =
+      await this.applicationService.getApplicationIdByParentApplicationId(
+        applicationId,
+      );
+    if (!currentApplicationId) {
+      throw new NotFoundException(
+        `Current application for application ${applicationId} was not found.`,
+      );
+    }
     return this.applicationControllerService.getApplicationProgressDetails(
-      applicationId,
+      currentApplicationId,
     );
   }
 
   /**
    * Get details for an application at 'Enrolment' status.
-   * @param applicationId student application id.
+   * @param applicationId student current application id.
    * @returns details for the application enrolment status.
    */
   @ApiNotFoundResponse({
     description:
-      "Application not found or not in relevant status to get enrolment details.",
+      "Application not found or current application for provided parent application not found or not in relevant status to get enrolment details.",
   })
   @Get(":applicationId/enrolment")
   async getEnrolmentApplicationDetails(
     @Param("applicationId", ParseIntPipe) applicationId: number,
   ): Promise<EnrolmentApplicationDetailsAPIOutDTO> {
+    const currentApplicationId =
+      await this.applicationService.getApplicationIdByParentApplicationId(
+        applicationId,
+      );
+    if (!currentApplicationId) {
+      throw new NotFoundException(
+        `Current application for application ${applicationId} was not found.`,
+      );
+    }
     return this.applicationControllerService.getEnrolmentApplicationDetails(
-      applicationId,
+      currentApplicationId,
     );
   }
 
   /**
    * Get details for an application at 'Completed' status.
-   * @param applicationId application id.
+   * @param applicationId current application id.
    * @returns details for an application on at completed status.
    */
   @ApiNotFoundResponse({
-    description: `Application not found or not on ${ApplicationStatus.Completed} status.`,
+    description: `Application not found or current application for provided parent application not found or not on ${ApplicationStatus.Completed} status.`,
```

Comment:
Are the API from [application-restriction-bypass.aest.controller.ts](https://github.com/bcgov/SIMS/blob/eea09edea675d14c255260ae076722e6775a6edd/sources/packages/backend/apps/api/src/route-controllers/application-restriction-bypass/application-restriction-bypass.aest.controller.ts#L60) not part of the PR?